cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -931,10 +952,9 @@ cdef class DeviceMemoryResource(MemoryResource):
             The size of the buffer to deallocate, in bytes.
         stream : Stream, optional
             The stream on which to perform the deallocation asynchronously.
-            If None, an internal stream is used.
+            If the buffer is deallocated without an explicit stream, the allocation stream
+            is used.
         """
-        if stream is None:
```

Comment:
Did we intend to remove this? Based on the `deallocate` API signature a user would deem it valid to pass `None` as a stream argument.  That `None` is not checked in the `_deallocate` function above when the stream object is called with it.